pkg/apis/core/validation/validation.go

```diff
@@ -4668,6 +4674,23 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		mungedPodSpec.TerminationGracePeriodSeconds = oldPod.Spec.TerminationGracePeriodSeconds // +k8s:verify-mutation:reason=clone
 	}
 
+	podIsGated := len(oldPod.Spec.SchedulingGates) > 0
+	if opts.AllowMutableNodeSelectorAndNodeAffinity && podIsGated {
```

Comment:
All updates to `PreferredDuringSchedulingIgnoredDuringExecution` are allowed too, I missed to add a comment on that.

Reply:
Added validation logic allowing any updates to PreferredDuringSchedulingIgnoredDuringExecution and a few test cases for this.